src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageDisplayer.cs

```diff
@@ -0,0 +1,422 @@
+                                LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_InvalidNuGetFeeds,
+                                packageToInstall,
+                                result.ErrorMessage).Bold().Red());
+                        break;
+                    case InstallerErrorCode.PackageNotFound:
+                        _reporterError.WriteLine(
+                            string.Format(
+                                LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_PackageNotFound,
+                                packageToInstall).Bold().Red());
+                        break;
+                    case InstallerErrorCode.DownloadFailed:
+                        _reporterError.WriteLine(
+                            string.Format(
+                                LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_DownloadFailed,
+                                packageToInstall).Bold().Red());
+                        break;
+                    case InstallerErrorCode.UnsupportedRequest:
+                        _reporterError.WriteLine(
+                            string.Format(
+                                LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_UnsupportedRequest,
+                                packageToInstall).Bold().Red());
+                        break;
+                    case InstallerErrorCode.AlreadyInstalled:
+                        _reporterError.WriteLine(
+                              string.Format(
+                                  LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_AlreadyInstalled,
+                                  packageToInstall).Bold().Red());
+                        _reporterError.WriteLine(LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_AlreadyInstalled_Hint, InstallCommand.ForceOption.Aliases.First());
+                        _reporterError.WriteCommand(Example.For<InstallCommand>(parseResult).WithArgument(BaseInstallCommand.NameArgument, packageToInstall).WithOption(BaseInstallCommand.ForceOption));
+
+                        break;
+                    case InstallerErrorCode.UpdateUninstallFailed:
+                        _reporterError.WriteLine(
+                              string.Format(
+                                  LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_UninstallFailed,
+                                  packageToInstall).Bold().Red());
+                        break;
+                    case InstallerErrorCode.InvalidPackage:
+                        _reporterError.WriteLine(
+                              string.Format(
+                                  LocalizableStrings.TemplatePackageCoordinator_lnstall_Error_InvalidPackage,
+                                  packageToInstall).Bold().Red());
+                        break;
+
+                    case InstallerErrorCode.VulnerablePackage:
+                        {
```

Comment:
you can do something like
var vulnerabilities = (result as InstallResult)?.Vulnerabilities ?? (result as UpdateResult)?.Vulnerabilities and remove duplicate code.

Reply:
depending on the command install/update we print different messages.
https://github.com/dotnet/sdk/pull/32002/files#:~:text=1mFailed%20to%20install%20testPackage%2C%20due%20to%20detected%20vulnerabilities

https://github.com/dotnet/sdk/pull/32002/files#:~:text=1mThe%20package%20testPackage%20wasn%27t

so I don't see the way to remove these pieces of code